### PR TITLE
[no-ticket][risk=no] Fix Spring-based command-line tools

### PR DESCRIPTION
**Pull request by dmohs**
Removes configuration garbage. Uses single source-of-truth for database configuration.

Ran set-authority and dev-up locally. Did not test every tool.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have added explanatory comments where the logic is not obvious
- [x] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/tools/src/main/java/org/pmiops/workbench/tools/Action.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.pmiops.workbench.tools;
+
+import org.pmiops.workbench.db.WorkbenchDbConfig;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+// This is a common ancestor that brings in the main database configuration.
+@Configuration
+@Import({WorkbenchDbConfig.class})
+public abstract class Action {}
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/Action.java:10]: nit: what about Tool or CommandLineTool, etc?

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
I appreciate the reduction of Spring magic in our codebase.

However I still cannot dev-up:

```
~/src/aou/api (dmohs/devup) $ ./project.rb dev-up
+ docker-compose up -d db
services.db Additional property env is not allowed
+ docker-compose down
services.db Additional property env is not allowed
~/src/aou/api (dmohs/devup) $
```

File: api/docker-compose.yaml [hunk 2]
```diff
@@ -3,8 +3,8 @@ version: "3.4"
 services:
   db:
     image: mariadb:10.2
-    env_file:
-      - db/local-vars.env
+    env:
```
Comment (jmthibault79) [inline api/docker-compose.yaml:6]: this needs to be `environment`.

Fixing that only brings me back to my recent problems however:
```
Connection could not be created to jdbc:mysql://db:3306/cdr with driver com.mysql.jdbc.Driver
```

**Comment by yonghaoy** [comment]
I saw `./run-migrations.sh: line 36: DB_HOST: unbound variable` in the failed `api-local-test` log

File: api/tools/src/main/java/org/pmiops/workbench/tools/Action.java (hunk 1 above)
Comment (dmohs) [reply api/tools/src/main/java/org/pmiops/workbench/tools/Action.java:10]: I was staring at SetAuthority and FetchFireCloudUserProfile and Action is what came to mind. I like Tool.

File: api/db/local-vars.env [hunk 3]
```diff
@@ -1,15 +1,10 @@
 # This is the vars.env file for running the local AoU API DB.
 # Cloud-SQL vars.env files are located in their projects' respective credentials buckets
 
-DB_DRIVER=com.mysql.jdbc.Driver
-# "db" hostname is established for the local docker network via docker-compose.yaml
-DB_HOST="${DB_HOST:-db}"
+DB_HOST=127.0.0.1
```
Comment (jmthibault79) [inline api/db/local-vars.env:4]: loving this simplification

**Review by jmthibault79 — COMMENTED** [review]
Seeing this after the 2:45 EDT push
```
+ echo 'Running liquibase '
Running liquibase
++ dirname ./generate-cdr/init-new-cdr-db.sh
+ cd ./generate-cdr/..
+ ../gradlew update -PrunList=schema

> Task :update FAILED
liquibase-plugin: Running the 'schema' activity...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/Users/thibault/.gradle/caches/modules-2/files-2.1/org.codehaus.groovy/groovy/2.4.12/a43be367110c491787219f1c128b5b5fc54f1e70/groovy-2.4.12.jar)
 to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Starting Liquibase at 14:50:36 (version 4.16.1 #4571 built at 2022-09-11 16:02+0000)
Liquibase Version: 4.16.1
WARNING: License service not loaded, cannot determine Liquibase Pro license status. Please consider re-installing Liquibase to include all dependencies. Continuing operation without Pro license.
[2023-03-31 14:50:37] SEVERE [liquibase.integration] Connection could not be created to jdbc:mysql://db:3306/cdr with driver com.mysql.jdbc.Driver.  Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
  - Caused by: db: nodename nor servname provided, or not known
liquibase.exception.CommandExecutionException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be crea
ted to jdbc:mysql://db:3306/cdr with driver com.mysql.jdbc.Driver.  Communications link failure
```

**Comment by dmohs** [comment]
@jmthibault79, I don't think this will resolve the issue you are seeing, but I do think it will get us closer to identifying the cause.

**Review by jmthibault79 — APPROVED** [review]
Does not fully resolve my issues with dev-up and tools (with my default gradle settings), but it seems to move us in the right direction.

If I disable the gradle daemon, I can run dev-up.

**Pull request merged**